modules/metastorage-client/src/main/java/org/apache/ignite/internal/metastorage/client/CursorImpl.java

```diff
@@ -125,13 +129,13 @@ public boolean hasNext() {
                             .get();
                 }
             } catch (InterruptedException | ExecutionException e) {
-                if (e.getCause() != null && e.getCause().getClass().equals(NodeStoppingException.class)) {
+                if (e.getCause() instanceof NodeStoppingException) {
                     return false;
                 }
 
                 LOG.debug("Unable to evaluate cursor hasNext command", e);
 
-                throw new IgniteInternalException(e);
+                throw withCause(MetaStorageException::new, CURSOR_EXECUTION_ERR, e);
```

Comment:
`e` can never be an Ignite*Exception. Do we determine its code by reading the message? Why don't we check the cause of ExecutionException?

Reply:
This cause is checked inside the `withCause` implementation:
```
    Throwable unwrapped = unwrapCause(t);
```
where `unwrap` if the following:
```
    public static Throwable unwrapCause(Throwable e) {
        while ((e instanceof CompletionException || e instanceof ExecutionException) && e.getCause() != null) {
            e = e.getCause();
        }

        return e;
    }
```